commit 5b2a955bdc5e4759f9f9d431d12dc2996be6e379 (refs/changes/62/4962/5)
Author: Dave Eddy <dave@daveeddy.com>
Date:   2018-10-18T14:43:04-04:00 (12 months ago)
    
    OS-7310 vminfod event stream processes events after stop called
    Reviewed by: Josh Wilsdon <josh@wilsdon.ca>
    Approved by: Josh Wilsdon <josh@wilsdon.ca>

diff --git a/src/vm/node_modules/vminfod/client.js b/src/vm/node_modules/vminfod/client.js
index aaac58a7..605cca78 100644
--- a/src/vm/node_modules/vminfod/client.js
+++ b/src/vm/node_modules/vminfod/client.js
@@ -482,6 +482,14 @@ VminfodEventStream.prototype._transform =
     var parsed;
     var vms;
 
+    if (!self.vs_req) {
+        // we are stopped - ignore any incoming data
+        self.vs_logger.trace({line: line},
+            'stream stopped - ignoring line from vminfod');
+        cb();
+        return;
+    }
+
     try {
         ev = JSON.parse(line);
         ev.date = new Date(ev.date);
